### ISSUE TITLE
Item fulfillment instead of Order fulfillment

### ISSUE THREAD

**Issue by Jaikant** [beckn/protocol-specifications#234]
Currently the specs has Order.fulfillment and Order.state

Instead of fulfillment being part of Order, there could be a fulfillment for each Item in the order. The completion of each item fulfillment would reflect in the Order.state. 

https://github.com/beckn/protocol-specifications/blob/b028fa3b8a5ea3c9be890ae673fa4472a4e78355/core/v0/api/core.yaml#L2063

**Comment by core-wg-admin** [comment]
@Jaikant that's a good idea! I'm assuming that you're talking about a multi-shipment order. Where for example, Item 1, 2, and 3 come in shipment 1 and then Item 4, 5, and 6 come in shipment 2 and so on.

I think adding the following fields to the `Item` schema should definitely help in the above scenaio.
1. `fulfillment_state` of type `string`. This could contain network-specified standard ENUMs to define the state of the fulfillment for that item 
2. `shipment_id` of type `string`. This could be set by the BPP any time during the Discovery, Order, or Fulfillment stages depending upon _when_ the shipment was created by the BPP. This field would contain a reference to the shipment that can be used by the BAP later to group items based on shipments.

@pramodkvarma any thoughts on the above?

**Comment by pramodkvarma** [comment]
Yes partial shipment and fulfilling orders across multiple shipments are very common B2B scenario. @venkatramanm is very familiar with these as well.
